src/server/transaction.cc

```diff
@@ -335,7 +335,7 @@ bool Transaction::RunInShard(EngineShard* shard) {
 
   // We make sure that we lock exactly once for each (multi-hop) transaction inside
   // transactions that lock incrementally.
-  if (incremental_lock && ((sd.local_mask & KEYLOCK_ACQUIRED) == 0)) {
+  if (!IsGlobal() && incremental_lock && ((sd.local_mask & KEYLOCK_ACQUIRED) == 0)) {
```

Comment:
Functional change 2

Reply:
I do not remember by heart the logic here, unfortunately. Can you explain the reason for this change?